### ISSUE TITLE
How to get the symbolic output of a layer for a symbolic input

### ISSUE THREAD

**Issue by stillbreeze** [fchollet/keras#3315]
Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

---

The loss function of my main model depends on outputs from a secondary network. So I need the symbolic outputs from certain layers of the secondary model when compiling my primary model.
The following code can give me the output for an intermediate layer if I have a numeric input:

`get_3_layer_output = K.function([model.layers[0].input, K.learning_phase()],[model.layers[3].output])
result = get_3_layer_output([X_train, 1])[0])`

Is there any way I can get the output for an intermediate layer with X_train being a symbolic variable instead of a numeric array?
